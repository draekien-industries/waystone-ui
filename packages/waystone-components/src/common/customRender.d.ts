/** @jsxImportSource theme-ui */
import React from 'react';
import { RenderOptions } from '@testing-library/react';
declare const DarkModeWrapper: React.FC;
declare const themedRender: (ui: React.ReactElement, options?: Omit<RenderOptions<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>, "wrapper"> | undefined) => import("@testing-library/react").RenderResult<typeof import("@testing-library/dom/types/queries"), HTMLElement, HTMLElement>;
export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
//# sourceMappingURL=customRender.d.ts.map