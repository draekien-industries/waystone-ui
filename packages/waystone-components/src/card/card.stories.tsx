import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Card, CardProps } from './card';
import exampleImage from './assets/example-image.png';

export default {
  title: 'Components/Card',
  component: Card,
  args: {
    children:
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Possimus harum dolor nemo atque reiciendis recusandae nam, voluptate earum? Sint, quae omnis! Enim labore harum reiciendis hic id. Eaque, dolores impedit?',
    fullWidth: false,
  },
} as Meta;

const Template: Story<CardProps> = (args) => <Card {...args} />;

export const Default = Template.bind({});

export const WithConstraints = Template.bind({});

WithConstraints.args = {
  maxWidth: 'md',
  maxHeight: 'md',
};

export const WithCoverImage = Template.bind({});

WithCoverImage.args = {
  coverImageUrl: exampleImage,
  maxWidth: 'xl',
};
