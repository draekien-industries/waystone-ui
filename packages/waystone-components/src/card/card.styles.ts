import { ThemeUIStyleObject } from 'theme-ui';
import { CanBeFullWidth, HasHeight, HasWidth } from '../common/interfaces';

/** THe props for styling the card container */
interface CardContainerCssProps extends HasWidth, HasHeight, CanBeFullWidth {
  /** Should the card not have a shadow */
  noShadow?: boolean;
  /** Is the app in dark mode. */
  darkMode?: boolean;
}

export const cardContainerCss = ({
  fullWidth,
  width,
  minWidth,
  maxWidth,
  noShadow,
  darkMode,
  ...rest
}: CardContainerCssProps): ThemeUIStyleObject => ({
  borderRadius: 'md',
  width: fullWidth ? '100%' : width || 'auto',
  minWidth: !fullWidth && minWidth,
  maxWidth: !fullWidth && maxWidth,
  backgroundColor: darkMode ? 'b-600' : 'b-000',
  boxShadow: !noShadow && 'md',
  overflowY: 'auto',
  ...rest,
});

export const coverImageCss = (coverImageUrl?: string): ThemeUIStyleObject => ({
  backgroundImage: coverImageUrl && `url(${coverImageUrl})`,
  backgroundColor: 'secondary',
  backgroundSize: 'cover',
  backgroundRepeat: 'no-repeat',
  backgroundPosition: 'center',
  width: '100%',
  height: 'md',
});

export const cardContentCss = (noPadding?: boolean): ThemeUIStyleObject => ({
  padding: noPadding ? 0 : 'md',
});
