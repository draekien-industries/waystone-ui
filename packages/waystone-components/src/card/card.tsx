/** @jsxImportSource theme-ui */

import { ThemeUIStyleObject, useColorMode } from 'theme-ui';
import { CanBeFullWidth, HasHeight, HasWidth } from '../common';

/**
 * The props for the Card component.
 */
export interface CardProps extends HasWidth, HasHeight, CanBeFullWidth {
  /** The children to render in the card. */
  children: React.ReactNode;
  /** Setting this prop to `true` will disable this component's drop shadow. */
  noShadow?: boolean;
}

/**
 * A card is a container for content that is grouped together.
 */
export const Card = ({
  children,
  fullWidth,
  width,
  minWidth,
  maxWidth,
  noShadow,
  ...rest
}: CardProps) => {
  const [mode] = useColorMode();
  const darkMode = mode === 'dark';

  const cardCss: ThemeUIStyleObject = {
    paddingX: 'md',
    paddingY: 'md',
    borderRadius: 'md',
    width: fullWidth ? '100%' : width || 'auto',
    minWidth: !fullWidth && minWidth,
    maxWidth: !fullWidth && maxWidth,
    backgroundColor: darkMode ? 'b-600' : 'b-000',
    boxShadow: !noShadow && 'md',
    overflowY: 'scroll',
    ...rest,
  };

  return <div sx={cardCss}>{children}</div>;
};
