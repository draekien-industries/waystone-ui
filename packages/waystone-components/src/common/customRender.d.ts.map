{"version":3,"file":"customRender.d.ts","sourceRoot":"","sources":["customRender.tsx"],"names":[],"mappings":"AAAA,gCAAgC;AAChC,OAAO,KAAoB,MAAM,OAAO,CAAC;AACzC,OAAO,EAAU,aAAa,EAAE,MAAM,wBAAwB,CAAC;AAQ/D,QAAA,MAAM,eAAe,EAAE,KAAK,CAAC,EAQ5B,CAAC;AAEF,QAAA,MAAM,YAAY,OACZ,MAAM,YAAY,sQAE8B,CAAC;AAEvD,cAAc,wBAAwB,CAAC;AACvC,OAAO,EAAE,YAAY,IAAI,MAAM,EAAE,eAAe,EAAE,CAAC"}