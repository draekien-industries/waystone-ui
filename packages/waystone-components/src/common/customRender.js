import { jsx as _jsx } from "theme-ui/jsx-runtime";
/** @jsxImportSource theme-ui */
import { useEffect } from 'react';
import { render } from '@testing-library/react';
import { Waystone } from '../';
import { useColorMode } from '@theme-ui/color-modes';
const TestProvider = ({ children }) => (_jsx(Waystone, { children: children }));
const DarkModeWrapper = ({ children }) => {
    const [, setColorMode] = useColorMode();
    useEffect(() => {
        setColorMode('dark');
    }, [setColorMode]);
    return _jsx("div", { children: children });
};
const themedRender = (ui, options) => render(ui, Object.assign({ wrapper: TestProvider }, options));
export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
