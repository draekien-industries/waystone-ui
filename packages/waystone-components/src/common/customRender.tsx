/** @jsxImportSource theme-ui */
import React, { useEffect } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { useColorMode } from 'theme-ui';
import { Waystone } from '../waystone-ui/waystoneUi';

const TestProvider: React.FC = ({ children }) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper: React.FC = ({ children }) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: TestProvider, ...options });

export * from '@testing-library/react';
export { themedRender as render, DarkModeWrapper };
