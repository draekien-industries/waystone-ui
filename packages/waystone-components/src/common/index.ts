export * from './customRender';
export * from './interfaces';
