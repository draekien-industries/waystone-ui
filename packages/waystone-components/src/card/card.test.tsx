import React from 'react';
import { render } from '../common';
import { Card } from './card';

describe('The Card component', () => {
  it('Matches snapshot', () => {
    const { container } = render(<Card>card</Card>);

    expect(container).toMatchSnapshot();
  });
});
