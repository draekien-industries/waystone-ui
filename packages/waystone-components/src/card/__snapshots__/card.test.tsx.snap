// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Card component Matches snapshot 1`] = `
<div>
  <div
    class="css-142j87o-Card"
  >
    card
  </div>
</div>
`;
